### DISCUSSION TITLE
ImportSTL: Invalid primitve error

### DISCUSSION THREAD

**Discussion by wishwah** [thliebig/openEMS-Project#69] [Q&A Usage] [answered]
Hi,
I am trying the ImportSTL function. I modelled an object on FreeCAD and imported the same but when I run, there are errors :

CSPrimPolyhedronReader::ReadFile: file invalid or empty, skipping ... 
CSPrimPolyhedronReader::ReadFromXML: Failed to read file. 
Warning: Invalid primitive found in property: dielectric2!

Why is this happening?What can I do to prevent the errors?Any help is much appreciated

**Comment by 0xCoto** [comment]
Could you check the mesh? It might be that the mesh lines aren't picking up the STL well enough and might require refinement.

**Reply by thliebig** [reply]
No this is clearly a file read error. If the mesh would not be okay, there would only be a warning about an unused object.

**Reply by 0xCoto** [reply]
Ah right, my bad.

**Comment by thliebig** [comment]
Did you export the STL as binary stl oder ASCII stl? Did you try to re-import the *.stl in FreeCAD or opened it with e.g. [meshlab](https://www.meshlab.net/)?
If the problem still exists, try to create a very simple object in FreeCAD (e.g. just a box) and try that one. If that fails to please attach the (ASCII) STL here?

**Reply by wishwah** [reply]
I exported the file as ASCII stl. I tried re-importing as well and there seemed to be no problem.Exporting a simple object (cube) failed as well.I am attaching the .txt file of the model I need to import to openEMS.
[cyl_plane.txt](https://github.com/thliebig/openEMS-Project/files/11515256/cyl_plane.txt)

**Reply by thliebig** [reply, marked as answer]
File looks fine. I guess I will need to give the import in openEMS a try...
Stupid question, you are sure openEMS can find the file? Try to set the full path as filename... Does the error message differ if it cannot find the file or is it the same message?

**Reply by wishwah** [reply]
You are right, openEMS could not find the file. I set the full path as the filename and there is no error in that case ! Thank you so much for the help Thorsten :)))
